src/views/v3/Bridge/Routes/index.tsx

```diff
@@ -140,79 +220,56 @@ function Routes({
     mobile ? setShowDrawer(false) : setShowModal(false);
   }, [highlightedRoute, toNativeToken, mobile, onRouteChange]);
 
-  const routesWithQuotes = routesList.filter((rs) => quotes[rs] !== undefined);
-
-  const fastestRoute = routesWithQuotes.reduce(
-    (fastest, route) => {
-      const quote = quotes[route];
-      if (!quote || !quote.success) return fastest;
-
-      if (
-        quote.eta !== undefined &&
-        quote.eta < fastest.eta &&
-        quote.eta < 60_000
-      ) {
-        return { name: route, eta: quote.eta };
-      }
-      return fastest;
-    },
-    { name: '', eta: Infinity },
-  );
-
-  const cheapestRoute = routesWithQuotes.reduce(
-    (cheapest, route) => {
-      const quote = quotes[route];
-      const rc = config.routes.get(route);
-      if (!quote || !quote.success || !rc.AUTOMATIC_DEPOSIT) return cheapest;
-
-      const amountOut = BigInt(quote.destinationToken.amount.amount);
-      if (amountOut > cheapest.amountOut) {
-        return { name: route, amountOut };
-      }
-      return cheapest;
-    },
-    { name: '', amountOut: 0n },
-  );
-
-  const selectedQuote = !selectedRoute
-    ? undefined
-    : (() => {
-        const quoteResult = quotes[selectedRoute];
-        return quoteResult?.success ? quoteResult : undefined;
-      })();
-
-  const bestRoute = fastestRoute.name
-    ? config.routes.get(fastestRoute.name)
-    : cheapestRoute.name
-    ? config.routes.get(cheapestRoute.name)
-    : undefined;
-
   const routeSection = useMemo(() => {
-    if (selectedRoute && selectedRoute !== bestRoute?.rc.meta.name) {
-      const route = config.routes.get(selectedRoute);
+    if (fastestRoute.name && cheapestRoute.name) {
       return (
-        <>
-          Route
-          {route.rc.meta.provider && (
-            <span
-              style={{ fontWeight: 500 }}
-            >{` via ${route.rc.meta.provider}`}</span>
-          )}
-        </>
-      );
-    } else {
-      return (
-        <>
-          Best route
-          {bestRoute?.rc.meta.provider && (
-            <span
-              style={{ fontWeight: 500 }}
-            >{` via ${bestRoute?.rc.meta.provider}`}</span>
-          )}
-        </>
+        <Box sx={{ maxWidth: '174px' }}>
+          <ToggleButtonGroup
+            exclusive
+            size="small"
+            value={selectedRouteBadge}
+            sx={styles.toggleGroup}
+            onChange={(_, value) => {
+              if (value === 'fastest' && fastestRoute.name) {
+                onRouteChange(fastestRoute.name);
+                setSelectedRouteBadge('fastest');
```

Comment:
See above comment, you may not need to set badge at all if onRouteChange already sets the route in state and passes it down to this component.